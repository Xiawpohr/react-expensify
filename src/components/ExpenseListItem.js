import React from 'react'
import { Link } from 'react-router-dom'
import moment from 'moment'
import numeral from 'numeral'
import { ListItem, ListItemTitle, ListItemSubtitle, ListItemData } from '../styles/List.js'

const ExpenseListItem = ({
  id,
  description,
  amount,
  createdAt
}) => {
  return id ? (
    <ListItem as={Link} to={`/edit/${id}`}>
      <div>
        <ListItemTitle as='h3'>{description}</ListItemTitle>
        <ListItemSubtitle>{moment(createdAt).format('MMMM Do, YYYY')}</ListItemSubtitle>
      </div>
      <ListItemData as='h3'>{numeral(amount / 100).format('$0,0.00')}</ListItemData>
    </ListItem>
  ) : (
    <ListItem message >
      <span>No expenses</span>
    </ListItem>
  )
}

export default ExpenseListItem
