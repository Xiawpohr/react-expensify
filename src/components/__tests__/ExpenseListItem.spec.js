import React from 'react'
import { shallow } from 'enzyme'
import ExpenseListItem from '../ExpenseListItem.js'

const expense = {
  id: '123',
  description: 'test expense',
  amuont: 100,
  createdAt: 0,
  note: 'test note'
}

test('should render ExpenseListItem correctly', () => {
  const wrapper = shallow(<ExpenseListItem {...expense} />)
  expect(wrapper).toMatchSnapshot()
})
