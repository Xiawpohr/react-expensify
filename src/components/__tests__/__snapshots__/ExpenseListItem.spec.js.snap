// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListItem correctly 1`] = `
<ListItem
  to="/edit/123"
>
  <div>
    <ListItemTitle
      as="h3"
    >
      test expense
    </ListItemTitle>
    <ListItemSubtitle>
      January 1st, 1970
    </ListItemSubtitle>
  </div>
  <ListItemData
    as="h3"
  >
    $0.00
  </ListItemData>
</ListItem>
`;
